src/azure-cli/azure/cli/command_modules/keyvault/custom.py

```diff
@@ -305,60 +305,6 @@ def list_vault(client, resource_group_name=None):
     return list(vault_list)
 
 
-def _get_current_user_object_id(graph_client):
-    current_user = graph_client.signed_in_user_get()
-    return current_user['id']
-
-
-def _get_object_id_by_spn(graph_client, spn):
-    accounts = list(graph_client.service_principal_list(
-        filter="servicePrincipalNames/any(c:c eq '{}')".format(spn)))
-    if not accounts:
-        logger.warning("Unable to find user with spn '%s'", spn)
-        return None
-    if len(accounts) > 1:
-        logger.warning("Multiple service principals found with spn '%s'. "
-                       "You can avoid this by specifying object id.", spn)
-        return None
```

Comment:
It is not possible for multiple service principals to have the same `spn`, so this check will never be hit. `azure.cli.command_modules.role.custom._resolve_object_id_and_type` has such logic:

https://github.com/Azure/azure-cli/blob/fbf4c610a5560dd9b5c01421ce3da87eb592b463/src/azure-cli/azure/cli/command_modules/role/custom.py#L1551